### ISSUE TITLE
Cannot build on Windows

### ISSUE THREAD

**Issue by blink1073** [rgbkrk/jupyter-sidecar#6]
I can `npm install` on Windows 7, but rebuilding `zmq` against the electron headers is a no go.  I get a ton of syntax errors from the VS project build, and this debug report.  Isn't Windows great?

```
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'build-zmq' ]
2 info using npm@2.10.1
3 info using node@v0.12.4
4 verbose run-script [ 'prebuild-zmq', 'build-zmq', 'postbuild-zmq' ]
5 info prebuild-zmq jupyter-sidecar@0.0.1
6 info build-zmq jupyter-sidecar@0.0.1
7 verbose unsafe-perm in lifecycle true
8 info jupyter-sidecar@0.0.1 Failed to exec build-zmq script
9 verbose stack Error: jupyter-sidecar@0.0.1 build-zmq: `cd node_modules/zmq && node-gyp rebuild --target=0.28.1 --arch=x64 --dist-url=https://atom.io/download/atom-shell`
9 verbose stack Exit status 1
9 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:213:16)
9 verbose stack     at EventEmitter.emit (events.js:110:17)
9 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
9 verbose stack     at ChildProcess.emit (events.js:110:17)
9 verbose stack     at maybeClose (child_process.js:1015:16)
9 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
10 verbose pkgid jupyter-sidecar@0.0.1
11 verbose cwd c:\Users\builder\jupyter-sidecar
12 error Windows_NT 6.1.7601
13 error argv "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build-zmq"
14 error node v0.12.4
15 error npm  v2.10.1
16 error code ELIFECYCLE
17 error jupyter-sidecar@0.0.1 build-zmq: `cd node_modules/zmq && node-gyp rebuild --target=0.28.1 --arch=x64 --dist-url=https://atom.io/download/atom-shell`
17 error Exit status 1
18 error Failed at the jupyter-sidecar@0.0.1 build-zmq script 'cd node_modules/zmq && node-gyp rebuild --target=0.28.1 --arch=x64 --dist-url=https://atom.io/download/atom-shell'.
18 error This is most likely a problem with the jupyter-sidecar package,
18 error not with npm itself.
18 error Tell the author that this fails on your system:
18 error     cd node_modules/zmq && node-gyp rebuild --target=0.28.1 --arch=x64 --dist-url=https://atom.io/download/atom-shell
18 error You can get their info via:
18 error     npm owner ls jupyter-sidecar
18 error There is likely additional logging output above.
19 verbose exit [ 1, true ]
```

**Comment by rgbkrk** [comment]
Couple prerequisities to make sure are on Windows:

* node-gyp needs to have access to Python 2, with a named `python` executable though there is a way to point node gyp to an alternative Python
* [zeromq headers for Windows](http://zeromq.org/distro:microsoft-windows)

**Comment by blink1073** [comment]
Yeah, I can build `zmq` just fine, just not the second time around.

**Comment by rgbkrk** [comment]
What happens when you cd to the zmq folder and run the electron build step:

```
HOME=~/.electron-gyp node-gyp rebuild --target=0.25.0 --arch=ia64 --dist-url=https://atom.io/download/atom-shell
```

Though I guess that HOME var would have to change to be Windowsy?

**Comment by blink1073** [comment]
gyp ERR! build error
gyp ERR! stack Error: ENOENT, open 'C:\Users\builder\.node-gyp\0.25.0\ia64\node.
lib'
gyp ERR! stack     at Error (native)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "node" "C:\\Users\\builder\\AppData\\Roaming\\npm\\node_modules
\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=0.25.0" "--arch=ia64" "--dist-
url=https://atom.io/download/atom-shell"
gyp ERR! cwd c:\Users\builder\jupyter-sidecar\node_modules\zmq
gyp ERR! node -v v0.12.4
gyp ERR! node-gyp -v v2.0.1
gyp ERR! not ok

**Comment by blink1073** [comment]
crap, just realized that `~` is not a valid path, let me try again

**Comment by rgbkrk** [comment]
If this next try doesn't help, it would be good to try a later version of node (e.g. iojs), though... it should be using Electron's node there...

**Comment by blink1073** [comment]
I used v0.12.4

**Comment by rgbkrk** [comment]
I wish we had node-zmq packaged as beautifully as pyzmq.

**Comment by rgbkrk** [comment]
Forever an issue.

**Comment by blink1073** [comment]
Can't they just start over from FreeBSD and save us all from misery?

**Comment by blink1073** [comment]
Is this still true @rgbkrk, didn't you release your own Windows compatible binary for this?

**Comment by rgbkrk** [comment]
Ha, yes this can be closed so thoroughly once we switch to `zeromq`. We ship prebuilds for all recent versions of Electron and node across Linux, macOS, and Windows.
